### DISCUSSION TITLE
Add API Support (Youtube, Twitter...) natively or addons/plugins

### DISCUSSION THREAD

**Discussion by leopoldhub** [maximmax42/Discord-CustomRP#42] [Ideas]
My suggestion is about adding API Support to update rich presence information with Youtube/Twitter... datas (subscribers, user logo, last tweet...)

user could add information about API keys/accounts in settings and use the data with special text syntax

Example:
`
Details:
[join my community of ${youtube.subscribers} members 😉]
`

If not integrated natively, the user could add custom APIs with addons/plugins.

**Comment by maximmax42** [comment]
Honestly this is a great idea, especially as addons/plugins, but I'm not skilled enough to do it. Maybe in the future, we'll see.

**Comment by cuboooooo** [comment]
You would have to have a way to import libraries right? like http requests and such. One way you could *use* APIs is by getting a random quote to cycle every 10 minutes, but maybe there could be an "advanced" tab to set the time and libraries up. 
https://zenquotes.io/api/random heres an API that spits out quotes, but its in json, so be sure to import json first if anyones willing to try.

**Reply by maximmax42** [reply]
My app already has a JSON library, RPC library is dependent on it.
